data/routers/postDb-router.js

```diff
@@ -0,0 +1,135 @@
+const router = require('express').Router();
+
+const postDb = require('../helpers/postDb');
+const userDb = require('../helpers/userDb');
+
+router.post('/', (req, res) => {
+  const newPost = req.body;
+  if (newPost.text && newPost.user_id) {
+    userDb.getById(newPost.user_id).then(user => {
+      if (user) {
+        postDb
+          .insert(newPost)
+          .then(post => {
+            res.status(201).json(post);
+          })
+          .catch(err => {
+            res.status(500).json({
+              error: 'There was an error while saving the post to the database'
+            });
+          });
+      } else {
+        res
+          .status(404)
+          .json({ error: 'The user with the specified ID does not exist.' });
+      }
+    });
+  } else {
+    res.status(400).json({
+      error: 'Please provide a valid user_id and text for the post.'
+    });
+  }
+});
```

Comment:
this is a deceptively involved endpoint but you accounted for all possible cases here, and throwing custom error messages everywhere possible as well--fantastic job. the next level of optimization is to use async/await in a try/catch block to clean it up and shorten your code, the same way arrow function syntax can clean up ES5 functions